File: CS/07_Attachments/GetPdfAttachmentInfo.cs
using Spire.Pdf;
using Spire.Pdf.Attachments;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace GetPdfAttachmentInfo
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create a new PDF document
            PdfDocument pdf = new PdfDocument();

            //Load the file from disk.
            pdf.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Pdf_2.pdf");

            //Get a collection of attachments on the PDF document
            PdfAttachmentCollection collection = pdf.Attachments;

            //Get the first attachment.
            PdfAttachment attachment = collection[0];

            //Get the information of the first attachment.
            StringBuilder content = new StringBuilder();
            content.AppendLine("Filename: " + attachment.FileName);
            content.AppendLine("Description: " + attachment.Description);
            content.AppendLine("Creation Date: " + attachment.CreationDate);
            content.AppendLine("Modification Date: " + attachment.ModificationDate);


            String result = "GetPdfAttachmentInfo_out.txt";

            //Save to file.
            File.WriteAllText(result, content.ToString());

            //Launch the file.
            DocumentViewer(result);
        }

        private void DocumentViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}

File: VB/07_Attachments/GetPdfAttachmentInfo.vb
Imports Spire.Pdf
Imports Spire.Pdf.Attachments
Imports System.IO
Imports System.Text

Namespace GetPdfAttachmentInfo
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new PdfDocument object
			Dim pdf As New PdfDocument()

			' Load the PDF document from the specified input file
			pdf.LoadFromFile("..\..\..\..\..\..\Data\Template_Pdf_2.pdf")

			' Get the collection of attachments in the PDF document
			Dim collection As PdfAttachmentCollection = pdf.Attachments

			' Access the first attachment in the collection
			Dim attachment As PdfAttachment = collection(0)

			' Create a StringBuilder to store attachment information
			Dim content As New StringBuilder()
			content.AppendLine("Filename: " & attachment.FileName)
			content.AppendLine("Description: " & attachment.Description)
			content.AppendLine("Creation Date: " & attachment.CreationDate)
			content.AppendLine("Modification Date: " & attachment.ModificationDate)

			' Specify the output file name for writing attachment information
			Dim result As String = "GetPdfAttachmentInfo_out.txt"

			' Write the attachment information to the output file
			File.WriteAllText(result, content.ToString())

			' Close the document
			pdf.Close()

			' Launch the file
			DocumentViewer(result)
		End Sub

		Private Sub DocumentViewer(ByVal filename As String)
			Try
				Process.Start(filename)
			Catch
			End Try
		End Sub
	End Class
End Namespace
